### ISSUE TITLE
Double scope declaration breaks tinymce.js

### ISSUE THREAD

**Issue by ldrick** [tinymce/tinymce#4165]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Since version 4.7.5 tinymce.js starts with a doubled scope declaration, like this:
```javascript
// 4.7.5 (2018-01-22)
(function () {
(function () {
// source code ...
}());
})()
```

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tinymce.com](http://fiddle.tinymce.com/) or similar.**
Just load tinymce@latest via yarn or npm and have a look at the node_modules/tinymce/tinymce.js

**What is the expected behavior?**
Have only one scope, or exporting everything according to 4.7.4

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**
It worked with 4.7.4 and before and currently effects only 4.7.5.
I ran into this one, due to I am loading tinymce via webpack and all my test fail trying to access anything from TinyMCE after loading the bundle.js.

**Comment by fyrkant** [comment]
Can you try installing tinymce nightly like this:
```sh
npm install http://download.ephox.com/tinymce/community/tinymce_nightly.tgz
```
And see if it working better?

**Comment by ldrick** [comment]
@fyrkant Thanks, this nightly build with
```javascript
(function () {
(function () {
    ...
    exportToModuleLoaders(Tinymce);
}());
})()
```
fixed tests and productive code on my side.

**Comment by fyrkant** [comment]
Great!

**Comment by ThomasPe** [comment]
This still seems broken?

**Comment by TheSpyder** [comment]
It's still doing this in TinyMCE 5, it's probably something weird in our bundle setup. I'll take a look early next year if I get some time.

**Comment by tuurbo** [comment]
Still broken as of 4.9.2

**Comment by PTC-JoshuaMatthews** [comment]
Any movement on this?

**Comment by TheSpyder** [comment]
Sorry, I forgot I had promised to follow up. After looking at it in more detail the extra function wrapping shouldn't matter. You can build TinyMCE without the double wrapping by removing the `banner` and `footer` properties on the rollup build to try that yourself:
https://github.com/tinymce/tinymce/blob/master/Gruntfile.js#L64

Whether it's double wrapped or not should be irrelevant, it's how the wrapped code exports itself that impacts integrations. Also if you use the `.min.js` file it's not double wrapped.

**Comment by lnewson** [comment]
I looked into the angular-cli issue more and it appears to be an issue in their logic. The cause appears to be that they are using metaprogramming to parse the imported modules (not entirely sure why, but it looks to be for processing enums), however their logic seems to be incorrect because as @TheSpyder said it's perfectly valid to have an iife inside another iife in JS (even though it is probably an edge case).

As such, I'd suggest logging an issue with the Angular team so that they can investigate/fix their processing logic.

**Comment by lnewson** [comment]
To add a little bit more detail, when TypeScript compiles an enum it ends up as JS something like this:
```
export var Direction;
(function (Direction) {
    Direction[Direction["Up"] = 0] = "Up";
})(Direction || (Direction = {}));
```
So what appears to be happening, is that the angular logic is incorrectly detecting the iife JS produced by TinyMCE to be an Enum and therefore saying it's invalid because it's not an enum.

**Comment by lnewson** [comment]
We've just merged a change to master that should fix this issue, as we've updated the build process so that we no longer require the double IIFE wrapper, that was making angular think TinyMCE was an enum declaration. That fix is currently scheduled to go out in 5.0.8.
